app/uk/gov/hmrc/incorporatedentityidentificationfrontend/testonly/stubs/controllers/CompaniesHouseStubController.scala

```diff
@@ -25,75 +26,123 @@ import scala.util.matching.Regex
+    companyNumber match {
+      case "00000001" => NotFound
+      case _ => Ok(Json.toJson(getCompanyProfile(companyNumber)))
+    }
+  }
 
-    val stubLtdCompanyProfileOverseas = Json.obj(
-      companyNameKey -> "Overseas LTD",
-      companyNumberKey -> companyNumber,
-      dateOfIncorporationKey -> stubOldDateOfIncorporation,
-      registeredOfficeAddressKey -> stubRegisteredOfficeAddressOverseas
-    )
+  private def getCompanyProfile(companyNumber: String): CompanyProfile =
+    companyNumber match {
+      case "12383404" => toFullCompanyProfile(companyNumber, companyNameSuffix = 4, address_line_1 = "10 HIGH STREET")
+      case "11132304" => toFullCompanyProfile(companyNumber, companyNameSuffix = 2, address_line_1 = "20 TEST DRIVE")
+      case "12404104" => toFullCompanyProfile(companyNumber, companyNameSuffix = 5, address_line_1 = "30 SPRING ROAD")
+      case "12423604" => toFullCompanyProfile(companyNumber, companyNameSuffix = 6, address_line_1 = "40 SUMMER ROW")
+      case "12444304" => toFullCompanyProfile(companyNumber, companyNameSuffix = 7, address_line_1 = "50 PAIGNTON PLACE")
+      case "12463604" => toFullCompanyProfile(companyNumber, companyNameSuffix = 8, address_line_1 = "60 ASTON AVENUE")
+      case "12482404" => toFullCompanyProfile(companyNumber, companyNameSuffix = 9, address_line_1 = "70 AUTUMN TERRACE")
+      case "12502004" => toFullCompanyProfile(companyNumber, companyNameSuffix = 10, address_line_1 = "80 GERRARD WAY").copy(company_name = "VAT PENALTY 10")
+      case "00000002" => CompanyProfile(
+        company_name = stubCompanyName,
+        company_number = companyNumber,
+        date_of_creation = stubOldDateOfIncorporation,
+        registered_office_address = Some(ukRegisteredOfficeAddress))
+      case "00000003" => CompanyProfile(
+        company_name = "Overseas LTD",
+        company_number = companyNumber,
+        date_of_creation = stubOldDateOfIncorporation,
+        registered_office_address = Some(overseasRegisteredOfficeAddress))
+      case CharitableIncorporatedOrganisation(_) => CompanyProfile(
+        company_name = "Test Charity",
+        company_number = companyNumber,
+        date_of_creation = "",
+        registered_office_address = None)
+      case _ => CompanyProfile(
+        company_name = stubCompanyName,
+        company_number = companyNumber,
+        date_of_creation = "2020-01-01",
+        registered_office_address = Some(ukRegisteredOfficeAddress))
+    }
 
-    val stubCioProfile = Json.obj(
-      companyNameKey -> stubCharityName,
-      companyNumberKey -> companyNumber,
-      dateOfIncorporationKey -> "",
-      registeredOfficeAddressKey -> Json.obj()
+  private def toFullCompanyProfile(companyNumber: String, companyNameSuffix: Int, address_line_1: String): CompanyProfile =
+    CompanyProfile(
+      company_name = "VAT PENALTY REFORM " + companyNameSuffix,
+      company_number = companyNumber,
+      date_of_creation = stubOldDateOfIncorporation,
```

Comment:
I think this is the wrong date of incorporation

Reply:
ops!